src/main/java/io/prismic/Api.java

```diff
@@ -313,38 +313,73 @@ public Ref getMaster() {
   }
 
   /**
-   * Retrieve multiple documents from their IDS
+   * Retrieve multiple documents from their ids on the given ref in the given language
+   */
+  public Form.SearchForm getByIDs(Iterable<String> ids, String ref, String lang) {
+    lang = lang != null ? lang : "*";
+    return this.query(Predicates.in("document.id", ids)).ref(ref).lang(lang);
+  }
+
+  /**
+   * Retrieve multiple documents from their ids on the given ref
+   */
+  public Form.SearchForm getByIDs(Iterable<String> ids, String ref) {
+    return this.getByIDs(ids, ref, null);
+  }
+
+  /**
+   * Retrieve multiple documents from their ids
    */
   public Form.SearchForm getByIDs(Iterable<String> ids) {
-    return this.query(Predicates.in("document.id", ids));
+    return this.getByIDs(ids, null, null);
   }
 
   /**
-   * Return the first document matching the predicate
+   * Return the first document matching the predicate on the given ref in the given language
    */
-  public Document queryFirst(Predicate p, String ref) {
+  public Document queryFirst(Predicate p, String ref, String lang) {
     if (ref == null) {
       ref = this.defaultReference == null ? this.getMaster().getRef() : this.defaultReference;
     }
-    List<Document> results = query(p).ref(ref).submit().getResults();
+    List<Document> results = query(p).ref(ref).lang(lang).submit().getResults();
     if (results.size() > 0) {
       return results.get(0);
     } else {
       return null;
     }
   }
 
+  /**
+   * Return the first document matching the predicate on the given reference
+   */
+  public Document queryFirst(Predicate p, String ref) {
```

Comment:
Missing the helper query which takes predicates + ref + lang and return List<Document>